src/main/java/de/unijena/cheminf/fragment/fingerprint/CountFingerprint.java

```diff
@@ -0,0 +1,179 @@
+ */
+public class CountFingerprint implements ICountFingerprint {
+    //<editor-fold desc="private  class variables" defaultstate="collapsed">
+    /**
+     * Is an array containing all predefined unique SMILES.
+     */
+    private String[] fragmentArrayOfUniqueSmiles;
+    /**
+     * The HashMap maps the unique SMILES to the position they have in the array.
+     */
+    private HashMap<String,Integer> uniqueSmilesToPositionMap;
+    /**
+     * Map
+     */
+    private HashMap<Integer, Integer> rawMap;
+    //</editor-fold>
+    //
+    //<editor-fold desc="Constructor" defaultstate="collapsed">
+    /**
+     * Constructor.
+     *
+     * @param aFragments
+     * @param aMapOfFragmentSmiles
+     */
+    public CountFingerprint(String[] aFragments, HashMap<String, Integer> aMapOfFragmentSmiles, HashMap<Integer, Integer> aRawMap) {
+        this.fragmentArrayOfUniqueSmiles = aFragments;
+        this.uniqueSmilesToPositionMap = aMapOfFragmentSmiles;
+        this.rawMap = aRawMap;
+    }
+    //</editor-fold>
+    //
+    //<editor-fold desc="Overridden public methods " defaultstate="collapsed">
+    /**
+     * Returns the number of bits in this fingerprint.
+     * Since fragment fingerprints are key based, the number of bits in the fingerprint
+     * is equal to the number of predefined fragments (unique SMILES).
+     *
+     * @return the size of the fingerprint.
+     */
+    @Override
+    public long size() {
+        return this.fragmentArrayOfUniqueSmiles.length;
+    }
+    //
+    /**
+     * Returns the number of bins that are populated.
+     * The number is equal to the defined fragments.
+     *
+     * @return the number of populated bins
+     * @see    #size()
+     */
+    @Override
+    public int numOfPopulatedbins() {
+        return this.fragmentArrayOfUniqueSmiles.length;
+    }
+    //
+    /**
+     * For each defined structure in the fingerprint, returns the frequency.
+     * For an index greater than or equal to 0 and less than fragmentArrayOfUniqueSmiles, the frequency of this unique SMILES is output.
+     * If the given index is greater than the size of the fingerprint, a NullPinterException is thrown.
+     *
+     * @param index the index of the bin to return the number of hits for.
+     * @return the count for the bin with given index.
+     */
+    @Override
+    public int getCount(int index) {
+        if (index < this.fragmentArrayOfUniqueSmiles.length && index >= 0 && this.rawMap.containsKey(index)) {
+            return this.rawMap.get(index);
+        } else if (index >= this.fragmentArrayOfUniqueSmiles.length) {
+            return this.rawMap.get(index);
+        } else {
+            return 0;
+        }
+    }
+    //
+    /**
+     * Returns the hash corresponding to the given index in the fingerprint.
+     * Since this is a key-based fingerprint, the hash value is nothing
+     * more than the position of the bin in the fingerprint (no hash value is calculated).
```

Comment:
Could it be that this method simply returns the number it was given, the index of an index?

Please remove "nothing more than".